### PR TITLE
UCT/IFACE - Bug fix, Wrong setting of post recv desc size and UD seg_size

### PR DESCRIPTION
**Pull request by ofirfarjun7**
## What
Fix bug - Wrong size of UD post receive buffers.
Fix inconsistency - seg_size should define the size of post recv buffers according to the description
Fix initialization of RX mpool elements size - can lead to confusion/memory waste in some transports (RC/DC).

## Why ?
Posting buffers with wrong size can cause unexpected behavior, buffer overrun when message is too big instead of "Not enough space" error code from HW
Fix UD seg size - Not meets seg_size definition, this creates a inconsistency in the meaning of seg_size between different transports.
Fix initialization of RX mpool elements size - Wrong element size can lead to confusion and a waste of memory.

## How ?
1. Make sure preparation of rx wqe's is done properly, with the right size.
2. Change UD seg_size to meet seg_size definition.
3. Change RX mpool element size accordingly.

### CODE REVIEW

File: src/uct/ib/rc/accel/rc_mlx5_common.c [hunk 1]
```diff
@@ -559,7 +559,7 @@ uct_rc_mlx5_common_iface_init_rx(uct_rc_mlx5_iface_common_t *iface,
     if (status != UCS_OK) {
         goto err;
     }
-
+
```
Comment (yosefe) [inline src/uct/ib/rc/accel/rc_mlx5_common.c:562]: pls revert

File: src/uct/ib/base/ib_iface.c [hunk 2]
```diff
@@ -261,7 +261,7 @@ ucs_status_t uct_ib_iface_recv_mpool_init(uct_ib_iface_t *iface,
     }
 
     return uct_iface_mpool_init(&iface->super, mp,
-                                iface->config.rx_payload_offset +
+                                iface->config.rx_hdr_offset +
```
Comment (yosefe) [inline src/uct/ib/base/ib_iface.c:264]: 👍 
since seg_size already includes the header

File: src/uct/ib/ud/base/ud_iface.c [hunk 3]
```diff
@@ -477,7 +477,7 @@ UCS_CLASS_INIT_FUNC(uct_ud_iface_t, uct_ud_iface_ops_t *ops,
     init_attr->rx_priv_len = sizeof(uct_ud_recv_skb_t) -
                              sizeof(uct_ib_iface_recv_desc_t);
     init_attr->rx_hdr_len  = UCT_IB_GRH_LEN + sizeof(uct_ud_neth_t);
-    init_attr->seg_size    = ucs_min(mtu, config->super.seg_size);
+    init_attr->seg_size    = ucs_min(mtu, config->super.seg_size) + UCT_IB_GRH_LEN;
```
Comment (yosefe) [inline src/uct/ib/ud/base/ud_iface.c:480]: added here because base IB code was fixed

File: src/uct/ib/ud/base/ud_iface.c [hunk 4]
```diff
@@ -739,9 +740,11 @@ ucs_status_t uct_ud_iface_query(uct_ud_iface_t *iface,
 
     iface_attr->cap.am.max_short       = uct_ib_iface_hdr_size(iface->config.max_inline,
                                                                sizeof(uct_ud_neth_t));
-    iface_attr->cap.am.max_bcopy       = iface->super.config.seg_size - sizeof(uct_ud_neth_t);
+    rx_hdr_len = iface->super.config.rx_payload_offset -
```
Comment (yosefe) [inline src/uct/ib/ud/base/ud_iface.c:743]: take rx_hdr_len directly
```
init_attr->rx_hdr_len = UCT_UD_RX_HDR_LEN
#define UCT_UD_RX_HDR_LEN UCT_IB_GRH_LEN + sizeof(uct_ud_neth_t);
```

**Comment by gleon99** [comment]
@ofirfarjun7 
Maybe rephrase / better explain in PR description:
"missing convention" -> "inconsistency".
"wrong size", "unexpected behavior" - try to be more specific and well-defined. Use terms like:
too big/small
Buffer overrun
**Undefined** behavior
Post message longer/shorter than ...
etc

File: src/uct/ib/ud/base/ud_iface.h [hunk 5]
```diff
@@ -29,7 +29,7 @@ BEGIN_C_DECLS
 
 
 #define UCT_UD_MIN_TIMER_TIMER_BACKOFF 1.0
-
+#define UCT_UD_RX_HDR_LEN (UCT_IB_GRH_LEN + sizeof(uct_ud_neth_t))
```
Comment (yosefe) [inline src/uct/ib/ud/base/ud_iface.h:32]: pls move to ud_def.h

**Review by yosefe — APPROVED** [review]

**Review by gleon99 — APPROVED** [review]
👍
